### DISCUSSION TITLE
View other user profile

### DISCUSSION THREAD

**Discussion by onlinejudge95** [fastapi-users/fastapi-users#1316] [Q&A]
I understand that I need to be a superuser If I am performing a write operation on other users, but for read operation it makes sense to be a non-superuser and still be able to perform the action. I am trying to implement a student-teacher relationship where the teacher can view the student's profile. Any pointers on the same would be helpful.

**Comment by hgalytoby** [comment]
You can customize the API to retrieve `users` and `users/{id}`.

**Reply by onlinejudge95** [reply]
I can write custom code for everything(mostly), I was just hoping if fastapi-users had this implemented or is this a design miss?
